spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Recurrent.scala

```diff
@@ -630,6 +659,25 @@ object Recurrent extends ContainerSerializable {
 
     createSerializeBigDLModule(recurrentBuilder, module)
   }
+
+  override def quantize[T: ClassTag](module: Module[T])(
+    implicit ev: TensorNumeric[T]): Module[T] = {
+    val recurrent = module.asInstanceOf[Recurrent[T]]
+    recurrent.clearState()
+
+    recurrent.topology = Quantizer.quantize(recurrent.topology).asInstanceOf[Cell[T]]
+    if (recurrent.preTopology != null) {
+      recurrent.preTopology = Quantizer.quantize(recurrent.preTopology)
+    }
+
+    recurrent.modules.clear()
+    if (recurrent.preTopology != null) {
+      recurrent.modules += recurrent.preTopology
+    }
+    recurrent.modules += recurrent.topology
+
+    recurrent
```

Comment:
This will modify the original layer, while in Linear and SpatialConvolution, the original layer is not modified.